### DISCUSSION TITLE
Trying to parse with Number.parseFloat() resulting in NaN value

### DISCUSSION THREAD

**Discussion by marcos-de-barros-vioque** [neuefische/questions#389] [Web] [answered]
Hello, 

After I could filter the data from the form in the `/addspending` page and map the entries on the corresponding category page (i.e. `/food`), I am now aiming for is a counter on each of the category pages (i.e. food/index.js) displaying the total amount of spendings for the corresponding category. The value should increase according to the "Amount" value indicated by the visitor through the form displayed on the `/addspending` page.

Therefore, I'm trying to filter and aggregate the data from the form using `.reduce()` and parse the value with the `Number.parseFloat()` method. However, **NaN** shows up instead of the correct aggregated value.

This is the code for the `SpendingsCounter` component:

```
export default function SpendingsCounter({
  spendingInput,
  amount,
  categoryName,
}) {
  let totalSpendingsValue = 0;
  let allSpendingsValues = 0;

  if (amount) {
    console.log(spendingInput, amount);
    spendingInput
      .filter(spendingInput => spendingInput.category === categoryName)
      .slice(amount)
      .reduce(amount => {
        allSpendingsValues = allSpendingsValues + Number.parseFloat(amount);
      });
  }

  totalSpendingsValue = amount + allSpendingsValues;

  return (
    <StyledSpendingsCounter>
      Total Amount: {totalSpendingsValue} €
    </StyledSpendingsCounter>
  );
}
```

Here the link to the PR: [https://github.com/marcos-de-barros-vioque/capstone/pull/24](https://github.com/marcos-de-barros-vioque/capstone/pull/24)

I would appreciate your help. Thanks a lot in advance.

**Comment by actyralf** [comment, marked as answer]
Hi Marcos,

you don't use reduce correctly: reduce returns the accumulated value (I think it's `allSpendingsValues`in your case). The callback you pass into reduce should return the previous value (usually called `accumulator` or `acc`) + the current value. Additionally I think you don't need `slice` here, right? Please try it this way:

```
const allSpendingsValues=spendingInput
      .filter(spendingInput => spendingInput.category === categoryName)
      .reduce((accumulator, currentSpending) => {
        return accumulator + Number.parseFloat(currentSpending.amount);
      },0);
```

I didn't test this with your code, but this is how reduce works.
Btw: `spendingInput` is not a good variable name, I suppose it's an array of spendings so you should consider to call it simply `spendings`. Use the plural form to indicate that it's an array.

**Reply by marcos-de-barros-vioque** [reply]
Thanks a lot, @actyralf, it's working now. And yes, I'll dedicate a future PR to adjust the namings throughout the project.
